### PR TITLE
Update aws_api_gateway.py

### PR DESCRIPTION
**Pull request by ToROxI**
Add ability to choose API Gateway by API name;
Add deletion of oldest stage when theirs count meets 10 (AWS`s limitation of stages count) and stage don't belong to the list of existing stages.

+label: docsite_pr

##### ISSUE TYPE

 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by zakabluk** [comment]
+

**Comment by ToROxI** [comment]
@samdoran , are you ok with such results of quality checks?
Please provide some additional information if I need to fix some issues on my side.

Thanks!

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/72027/66/tests

**Comment by ToROxI** [comment]
@mattclay ,
Hi!
Could I fix this issue from my side?

**Comment by mattclay** [comment]
@ToROxI The current integration test failure is here:

https://app.shippable.com/github/ansible/ansible/runs/72182/66/tests

It looks like the main failure is due to a traceback in the module:

```
Traceback (most recent call last):
  File "/tmp/ansible_hpr2UW/ansible_module_aws_api_gateway.py", line 407, in <module>
    main()
  File "/tmp/ansible_hpr2UW/ansible_module_aws_api_gateway.py", line 200, in main
    deploy_desc=deploy_desc)
  File "/tmp/ansible_hpr2UW/ansible_module_aws_api_gateway.py", line 284, in ensure_api_in_correct_state
    description=deploy_desc)
  File "/tmp/ansible_hpr2UW/ansible_modlib.zip/ansible/module_utils/cloud.py", line 153, in retry_func
IndexError: list index out of range
```

Can you run the tests locally to reproduce the issue?

Assuming you're running from the devel branch and have already sourced `hacking/env-setup` you can run the tests locally with: `ansible-test integration aws_api_gateway --docker -v`

You'll need to provide AWS credentials to run the tests.

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 1]
```diff
@@ -33,6 +33,10 @@
   api_id:
     description:
       - The ID of the API you want to manage.
+  api_name:
+    description:
+      - Name of the API you want to manage.
+    version_added: '2.7'
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py:39]: `'2.8'`

**Comment by willthames** [comment]
This change needs tests and exception handling improvements (I realise the original exception handling is lacking) - see https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#exception-handling. Other than that the general approach seems sound and valid.

**Comment by gundalow** [comment]
@ToROxI @willthames 
Thank you for your PR.
As part of reviewing the backlog of PRs we are looking at PRs older PRs that haven't been updated in a while

Given that:
* PR has failed CI over a year
* This is a feature rather than a bug fix
* The source branch (in your fork) no longer exists

Therefore I'm going to close this.

If you or anyone else wants to continue with this work then please do feel free to create a fresh PR and `@mention` the previous reviewers in here.

**Pull request closed without merging**